lib/Arch/Sleigh/PPCArch.cpp

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2022-present Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PPC.h"
+
+#define INCLUDED_FROM_REMILL
+#include <remill/Arch/PPC/Runtime/State.h>
+
+namespace remill {
+
+namespace sleighppc {
+
+static const char *kPPCVLERegName = "VLEReg";
+
+SleighPPCDecoder::SleighPPCDecoder(const remill::Arch &arch)
+    : SleighDecoder(arch, "ppc_64_isa_vle_be.sla", "ppc_64.pspec",
+                    {{"vle", kPPCVLERegName}}, {}) {}
+
+llvm::Value *SleighPPCDecoder::LiftPcFromCurrPc(llvm::IRBuilder<> &bldr,
+                                                llvm::Value *curr_pc,
+                                                size_t curr_insn_size) const {
+  // PC on thumb points to the next instructions next.
+  return bldr.CreateAdd(
+      curr_pc, llvm::ConstantInt::get(curr_pc->getType(), curr_insn_size));
+}
+
+void SleighPPCDecoder::InitializeSleighContext(
+    remill::sleigh::SingleInstructionSleighContext &ctxt,
+    const std::map<std::string, uint64_t> &context_values) const {
+  // If the context value mappings specify a value for the VLE register, let's pass that into
+  // Sleigh.
+  //
+  // Otherwise, default to VLE off.
+  uint64_t vle_value = 0;
+  const auto vle_iter = context_values.find(kPPCVLERegName);
```

Comment:
Extract this as standalone helper function since is sounds generically useful (both get/set with defaults).

Reply:
[a689327](https://github.com/lifting-bits/remill/pull/645/commits/a6893274aad322ae67bac72af8c28cfc52a966d0)